CA1508: False positive in case of declaration patterns

### Analyzer

**Diagnostic ID**: CA1508 (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code

### Steps To Reproduce

Compile the following code fragment:

```cs
public class Runner
{
    public string Run(object value)
    {
        if (value is null)
        {
            return null;
        }

        var x =  value is double num1;
        if (x)
        {
            return num1.ToString(CultureInfo.InvariantCulture);
        }

        if (!(value is int num2))
        {
            return null;
        }

        return num2.ToString(CultureInfo.InvariantCulture);
    }
}
```

### Expected behavior

No diagnostics are reported.

### Actual behavior

The following diagnostics are reported:

```text
Runner.cs(12,18,12,38): error CA1508: 'value is double num1' is always 'true'. Remove or refactor the condition(s) to avoid dead code.
Runner.cs(18,15,18,32): error CA1508: 'value is int num2' is always 'true'. Remove or refactor the condition(s) to avoid dead code.
```

If you remove the null check from the method, then CA1508 is no longer reported.

Note that the following diagnostic is also reported (but that is another issue):
Runner.cs(15,20,15,24): error CS0165: Use of unassigned local variable 'num1'

--- a/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_ValueContentAnalysis.cs
+++ b/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_ValueContentAnalysis.cs
@@ -3275,5 +3275,36 @@ class C
                 LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8
             }.RunAsync();
         }
+
+        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.ValueContentAnalysis)]
+        [Fact, WorkItem(6483, "https://github.com/dotnet/roslyn-analyzers/issues/6483")]
+        public async Task IsPatternExpression_Unboxing_NoDiagnosticsAsync()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    public void Run(object value)
+    {
+        if (value is null)
+        {
+            return;
+        }
+
+        var x = value is double num1;
+        if (x)
+        {
+            return;
+        }
+
+        if (!(value is int num2))
+        {
+            return;
+        }
+    }
+}",
+            }.RunAsync();
+        }
     }
 }

